compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -15247,7 +15247,7 @@ private boolean isDefiniteTypeDesc(SyntaxKind kind) {
     }
 
     private boolean isDefiniteExpr(SyntaxKind kind) {
-        if (kind == SyntaxKind.QUALIFIED_NAME_REFERENCE || kind == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+        if (kind == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
```

Comment:
Let's not modify the `isDefiniteExpr()`. As the method name implies this should return true only for the constructs that are definitely an expression. Both `SIMPLE_NAME_REFERENCE` and `QUALIFIED_NAME_REFERENCE` can be type references as well.

Reply:
Updated. ab6de6ea10bbdedd98eeccca0f40b5cbcdbb1900